modules/azure/aks.go

```diff
@@ -0,0 +1,72 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-11-01/containerservice"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/retry"
+)
+
+// GetManagedClustersClient is a helper function that will setup an Azure ManagedClusters client on your behalf
+func GetManagedClustersClient(subscriptionID string) (*containerservice.ManagedClustersClient, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	managedServicesClient := containerservice.NewManagedClustersClient(subscriptionID)
+	authorizer, err := NewAuthorizer()
+
+	if err != nil {
+		return nil, err
+	}
+
+	managedServicesClient.Authorizer = *authorizer
+
+	return &managedServicesClient, nil
+}
+
+// GetManagedCluster will return ContainerService
+func GetManagedCluster(t *testing.T, resourceGroupName, clusterName, subscriptionID string) (*containerservice.ManagedCluster, error) {
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	client, err := GetManagedClustersClient(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	managedCluster, err := client.Get(context.Background(), resourceGroupName, clusterName)
+	if err != nil {
+		return nil, err
+	}
+	return &managedCluster, nil
+}
+
+// WaitUntilServiceExternalIPsAvailable is waiting for allocation of External IP Address
+func WaitUntilServiceExternalIPsAvailable(t *testing.T, options *k8s.KubectlOptions, serviceName string, retries int, sleepBetweenRetries time.Duration) {
```

Comment:
This already exists in `k8s`: https://github.com/gruntwork-io/terratest/blob/master/modules/k8s/service.go#L58

Reply:
@yorinasub17 It doesn't work for Azure. In case of Azure, we need to use LoadBalancer or ingress type to accept external access. For the Load Balancer with Azure, the method go to the next line if it creates a service, however, we need to wait, until the service is actually get an IP Address. That is why, I create this method. The next question is should I keep it on Azure or should I move it to k8s namespace? If it is general, k8s might be useful for users, however, if it happens only on Azure, I'm wondering where is a good place to put this method.

Reply:
Hmm that seems like a bug if you are seeing this behavior, because it should be doing exactly that: wait for IP address

Can you see what is wrong with https://github.com/gruntwork-io/terratest/blob/master/modules/k8s/service.go#L81?

Reply:
@yorinasub17  I read the code, it looks exactly the same as mine. 
I suspect that the code is the same, so I remove mine, and point k8s namespace one. 
Maybe the problem is, Even after the IPAddress is allocated, it looks still take time to fetch it.
My test execute polling after the IPAddress readiness, so it might hide the issue. 
 
I'm testing my method ,then I'll replace the method to k8s's and replace it then observe the result, if it looks the same, I'm remove my method.

Reply:
Hi @yorinasub17 
I figure out what is the problem.  It is not about the `k8s.WaitUntilServiceAvailable()` It works well, however, the problem is, `k8s.GetServiceEndpoint()` The endpoint returns an empty string in case of LoadBalancer type of Azure. Since the method reference Hostname. 

![image](https://user-images.githubusercontent.com/1390976/73234201-f35d2600-413d-11ea-9977-76a3d2f67d18.png)

The GetServiceEndpointE's documentation said that it return hostname, the behavior works correctly.  

I'm gonna remove WaitFor... method and use `k8s`  and try not to use `GetServiceEndpoint` for Azure.  I'll use 

```go 
	endpoint := service.Status.LoadBalancer.Ingress[0].IP
```
  instead.

Reply:
Absolutely, I've done it. Please make sure run CI test for Kubernetes.